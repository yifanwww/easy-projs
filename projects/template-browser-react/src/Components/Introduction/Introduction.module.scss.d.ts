declare const styles: {
    introduction: string;
    link: string;
    logo: string;
};
export = styles;
