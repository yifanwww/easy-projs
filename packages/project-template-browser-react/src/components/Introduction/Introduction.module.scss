.introduction {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 60vh;
    padding-top: 20vh;
    font-size: calc(10px + 2vmin);
}

.link {
    color: #61dafb;
}

.logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .logo {
        animation: logo-spin infinite 20s linear;
    }
}

@keyframes logo-spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}
