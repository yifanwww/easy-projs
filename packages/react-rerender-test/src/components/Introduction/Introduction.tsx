import Assets from '@easy/assets';
import { ReactElement } from 'react';

import scss from './Introduction.module.scss';

export function Introduction(): ReactElement {
    return (
        <header className={scss.introduction}>
            <img className={scss.logo} src={Assets.logo} alt="logo" />
            <p>
                Edit <code>src/components/Introduction/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={scss.link}
                href="https://github.com/yifanwww/easy-projs"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn react-rerender-test
            </a>
        </header>
    );
}
