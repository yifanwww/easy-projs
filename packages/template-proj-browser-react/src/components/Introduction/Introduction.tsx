import Assets from '@easy/assets';

import scss from './Introduction.module.scss';

export const Introduction: React.FC = () => {
    return (
        <header className={scss.introduction}>
            <img className={scss.logo} src={Assets.logo} alt="logo" />
            <p>
                Edit <code>src/components/Introduction/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={scss.link}
                href="https://github.com/yifanwww/easy-projs"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn easy-projs
            </a>
        </header>
    );
};
