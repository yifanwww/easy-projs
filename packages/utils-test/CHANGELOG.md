# CHANGELOG
## @packages/utils-test v0.2.0 (Unreleased)
### Features

- Add `mockLocalStorage` and `mockSessionStorage` to help mock local storage and session storage
- Add test utils: `expectElementSnapshot`, `expectSnapshot`, `mockConsoleToMatchSnapshot`

## @packages/utils-test v0.1.0 (2021-10-29)
### Features

- Add `test.setup.ts` to setup jest globally for every package
- Add `validateHookValueNotChanged` to help test React hooks
