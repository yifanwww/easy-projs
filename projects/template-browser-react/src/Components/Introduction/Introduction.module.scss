.introduction {
    align-items: center;
    background: #282c34;
    color: white;
    display: flex;
    flex-direction: column;
    font-size: calc(10px + 2vmin);
    justify-content: center;
    min-height: 100vh;
}

.link {
    color: #61dafb;
}

.logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .logo {
        animation: App-logo-spin infinite 20s linear;
    }
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
