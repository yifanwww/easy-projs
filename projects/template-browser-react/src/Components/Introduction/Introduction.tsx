import { ReactElement } from 'react';

import logo from './logo.svg';
import scss from './Introduction.module.scss';

export function Introduction(): ReactElement {
    return (
        <header className={scss.introduction}>
            <img className={scss.logo} src={logo} alt="logo" />
            <p>
                Edit <code>src/Components/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={scss.link}
                href="https://github.com/yifanwww/easy-projs"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn easy-projs
            </a>
        </header>
    );
}
