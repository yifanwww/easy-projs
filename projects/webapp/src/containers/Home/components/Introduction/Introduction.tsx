import { Assets } from 'src/assets';

import css from './Introduction.module.scss';

export function Introduction() {
    return (
        <header className={css.introduction}>
            <img className={css.logo} src={Assets.logo} alt="logo" />
            <p>
                Edit <code>src/containers/Home/components/Introduction/Introduction.tsx</code> and save to reload.
            </p>
            <a
                className={css.link}
                href="https://github.com/yifanwww/easy-projs"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn easy-projs
            </a>
        </header>
    );
}
