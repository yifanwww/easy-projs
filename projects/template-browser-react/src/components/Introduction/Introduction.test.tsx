import { expectElementSnapshot } from '@easy-pkg/utils-test';
import { render } from '@testing-library/react';

import { Introduction } from './Introduction';

describe('Test component `Introduction`', () => {
    it('renders', () => {
        const reactElement = <Introduction />;

        expectElementSnapshot(reactElement);

        const { getByText } = render(reactElement);
        const linkElement = getByText(/Learn easy-projs/i);
        expect(linkElement).toBeInTheDocument();
    });
});
