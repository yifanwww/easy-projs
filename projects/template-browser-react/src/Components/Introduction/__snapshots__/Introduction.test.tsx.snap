// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component \`Introduction\` renders 1`] = `
<header
  className="introduction-54"
>
  <img
    alt="logo"
    className="logo-56"
    src="logo.svg"
  />
  <p>
    Edit 
    <code>
      src/renderer/MainWindow/App.tsx
    </code>
     and save to reload.
  </p>
  <a
    className="link-55"
    href="https://github.com/YSoftwareRepo/template-electron-cra"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn @YSoftwareRepo/template-electron-cra
  </a>
</header>
`;
